src/Microsoft.AspNet.Mvc.Core/DefaultControllerActivator.cs

```diff
@@ -3,106 +3,27 @@
+        private static readonly Func<Type, CreateControllerThunk> _createFactoryThunk =
+            type => ActivatorUtilities.CreateFactory(type, Type.EmptyTypes);
+        private readonly ConcurrentDictionary<Type, CreateControllerThunk> _controllerThunks =
+            new ConcurrentDictionary<Type, CreateControllerThunk>();
 
-            for (var i = 0; i < propertiesToActivate.Length; i++)
-            {
-                var activateInfo = propertiesToActivate[i];
-                activateInfo.Activate(controller, context);
-            }
-        }
-
-        protected virtual IDictionary<Type, Func<ActionContext, object>> CreateValueAccessorLookup()
-        {
-            var dictionary = new Dictionary<Type, Func<ActionContext, object>>
-            {
-                { typeof(ActionContext), (context) => context },
-                { typeof(HttpContext), (context) => context.HttpContext },
-                { typeof(HttpRequest), (context) => context.HttpContext.Request },
-                { typeof(HttpResponse), (context) => context.HttpContext.Response },
-                {
-                    typeof(ViewDataDictionary),
-                    (context) =>
-                    {
-                        var serviceProvider = context.HttpContext.RequestServices;
-                        return new ViewDataDictionary(
-                            serviceProvider.GetRequiredService<IModelMetadataProvider>(),
-                            context.ModelState);
-                    }
-                },
-                {
-                    typeof(ActionBindingContext),
-                    (context) =>
-                    {
-                        var serviceProvider = context.HttpContext.RequestServices;
-                        var accessor = serviceProvider.GetRequiredService<IScopedInstance<ActionBindingContext>>();
-                        return accessor.Value;
-                    }
-                }
-            };
-            return dictionary;
-        }
-
-        private PropertyActivator<ActionContext> CreateActivateInfo(
-            PropertyInfo property)
+        /// <inheritdoc />
+        public object Create([NotNull] ActionContext actionContext, [NotNull] Type controllerType)
         {
-            Func<ActionContext, object> valueAccessor;
-            if (!_valueAccessorLookup.TryGetValue(property.PropertyType, out valueAccessor))
-            {
-                valueAccessor = (actionContext) =>
-                {
-                    var serviceProvider = actionContext.HttpContext.RequestServices;
-                    return serviceProvider.GetRequiredService(property.PropertyType);
-                };
-            }
-
-            return new PropertyActivator<ActionContext>(property, valueAccessor);
+            var thunk = _controllerThunks.GetOrAdd(controllerType, _createFactoryThunk);
+            return thunk(actionContext.HttpContext.RequestServices, null);
```

Comment:
I think not - at bare minimum, it's supposed to be equivalent to `Activator.CreateInstance` \ `new T()`